mri/reconstructors/tests/test_reconstructors.py

```diff
@@ -0,0 +1,249 @@
+                dimension=len(fourier.shape),
+                nb_scale=2,
+                n_coils=self.num_channels,
+            )
+            # For self calibrating reconstruction the n_coils
+            # for wavelet operation is 1
+            linear_op.n_coils = 1
+            reconstructor = SelfCalibrationReconstructor(
+                fourier_op=fourier,
+                linear_op=linear_op,
+                gradient_method=formulation,
+                lips_calc_max_iter=num_iter,
+                verbose=0,
+            )
+            x_final, costs, _ = reconstructor.reconstruct(
+                kspace_data=kspace_data,
+                optimization_alg=optimizer,
+                num_iterations=num_iter,
+            )
+            # TODO add checks on result
+            # This is just an integration test, we dont have any checks for
+            # now. Please refer to tests for extracting sensitivity maps
+            # for more tests
+
+    def test_sparse_calibrationless_reconstruction(self):
+        """ Test all the registered transformations.
+        """
+        self.num_channels = 2
+        num_iter = 2
+        print("Process test for SparseCalibrationlessReconstructor ::")
+        for i in range(len(self.test_cases)):
+            print("Test Case " + str(i) + " " + str(self.test_cases[i]))
+            image, nb_scale, optimizer, recon_type, name = self.test_cases[i]
+            image_multichannel = np.repeat(image.data[np.newaxis],
+                                           self.num_channels, axis=0)
+            if optimizer == 'condatvu':
+                formulation = "analysis"
+            else:
+                formulation = "synthesis"
+            if recon_type == 'cartesian':
+                fourier = FFT(
+                    samples=convert_mask_to_locations(self.mask),
+                    shape=image.shape,
+                    n_coils=self.num_channels)
+            else:
+                fourier = NonCartesianFFT(
+                    samples=convert_mask_to_locations(self.mask),
+                    shape=image.shape,
+                    n_coils=self.num_channels)
+            kspace_data = fourier.op(image_multichannel)
+            linear_op = self.get_linear_operator(
+                wavelet_name=name,
+                dimension=len(fourier.shape),
+                nb_scale=2,
+                n_coils=2,
+                n_jobs=2,
+            )
+            reconstructor = SparseCalibrationlessReconstructor(
+                fourier_op=fourier,
+                linear_op=linear_op,
+                gradient_method=formulation,
+                lips_calc_max_iter=num_iter,
+                verbose=0,
+            )
+            x_final, costs, _ = reconstructor.reconstruct(
+                kspace_data=kspace_data,
+                optimization_alg=optimizer,
+                num_iterations=num_iter,
+            )
+            # TODO add checks on result
```

Comment:
I'm sorry but I can not see the reason for that ?

Reply:
The multichannel case with both having full Smaps, we cant have a check like earlier for 2 reasons:
1) We have only 2 iterations here as we dont want to test all cases in multichannel which would terribly slow down the test run time. As pointed, this is a mere integration test at this point.
2) I am not very sure about this, but the solution in this case will not exactly match the inverse fourier transform, but rather the square root of sum of squares of it. 

Still, given that these tests didnt exist at all, and that we are revamping codebase , I felt it was right point to add basic integration tests that just ensures we can run things.

Reply:
> 1. We have only 2 iterations here as we dont want to test all cases in multichannel which would terribly slow down the test run time. As pointed, this is a mere integration test at this point.

You can still synthesize  low dimension data with only 2 channel (just avoid even repeated entry)

> 2. I am not very sure about this, but the solution in this case will not exactly match the inverse fourier transform, but rather the square root of sum of squares of it.

The problem is now entirely separable which means that it is now equivalent of solving each channel independently. Just do not only test the SOS 
> 
> Still, given that these tests didnt exist at all, and that we are revamping codebase , I felt it was right point to add basic integration tests that just ensures we can run things.

Reply:
I have added checks here based on image model. Checks are placed on each channel for calibrationless and on SOS for self calibrating.